File: Hackerrank/AdInfinitum/akhil_gf.py
t = (int)(input())

while t != 0 :
	N,M = [int(x) for x in input().split()]
	sum = 0
	count = 0
	while count != N:
		sum+=10**count
		count+=1
	print(sum)
	print(sum%M)
	t-=1
	

File: Hackerrank/AdInfinitum/akhil_gf.cpp
#include<iostream>
#include<math.h>
typedef long long int LL;
using namespace std;

int main()
{
	long t;
	cin>>t;
	while( t-- )
	{
		LL N,M;
		cin>>N>>M;
		LL sum = 0;
		LL count = 0;
		while( count != N )
		{
			sum = sum + (LL)pow((double)10,count);
			count++;
		
		}
		
		cout<<sum%M<<endl;
	}
	
	return 0;

}